lib/rules/no-underscore-dangle.js

```diff
@@ -126,16 +131,40 @@ module.exports = {
         }
 
         /**
-         * Check if function has a underscore at the end
+         * Check if function parameter has a dangling underscore.
+         * @param {ASTNode} node node to evaluate
+         * @returns {boolean} true if function parameter name has a dangling underscore.
+         * @private
+         */
+        function checkForDanglingUnderscoreInFunctionParameter(node) {
+            if (!allowFunctionParams) {
+                node.params.forEach(param => {
+                    const identifier = param.name;
+
+                    if (typeof identifier !== "undefined" && hasDanglingUnderscore(identifier)) {
```

Comment:
I think we should allow identifiers specified in `allow option` using with `isAllowed()` function.
And It would be great if we can add some test cases for ensuring it.